pennylane/utils.py

```diff
@@ -19,24 +19,63 @@
 from collections.abc import Iterable
 from collections import OrderedDict
 import copy
-import numbers
 import functools
 import inspect
 import itertools
 
 import numpy as np
 
 import pennylane as qml
-from pennylane.variable import Variable
+
+
+def is_iterable(x):
+    """Predicate for deciding if an object is iterable, in the sense of :func:`_flatten`.
+
+    `str` and `bytes` are not considered iterable here.
+
+    Args:
+        x (Any): object
+    Returns:
+        bool: True iff x is iterable
+    """
+    return isinstance(x, Iterable) and not isinstance(x, (str, bytes))
```

Comment:
Agreed. Note that `is_iterable` could and maybe should be used in _flatten and _unflatten as well, but I did not make this change yet because the function call overhead might affect the performance. _flatten does get called a lot.